daemon/containerd/image_history.go

```diff
@@ -18,31 +17,41 @@ import (
 // ImageHistory returns a slice of HistoryResponseItem structures for the
 // specified image name by walking the image lineage.
 func (i *ImageService) ImageHistory(ctx context.Context, name string) ([]*imagetype.HistoryResponseItem, error) {
-	desc, err := i.resolveDescriptor(ctx, name)
+	desc, err := i.resolveImage(ctx, name)
 	if err != nil {
 		return nil, err
 	}
 
 	cs := i.client.ContentStore()
-	// TODO: pass the platform from the cli
-	conf, err := containerdimages.Config(ctx, cs, desc, platforms.AllPlatformsWithPreference(cplatforms.Default()))
-	if err != nil {
-		return nil, err
-	}
+	// TODO: pass platform in from the CLI
+	platform := platforms.AllPlatformsWithPreference(cplatforms.Default())
 
-	diffIDs, err := containerdimages.RootFS(ctx, cs, conf)
+	var manifests []ocispec.Manifest
```

Comment:
This seems to be largely the same as in `image.go` -- is this generic enough to hoist into a helper?

Reply:
I looked around and didn't see anywhere else where this would make sense (for now), so I'll leave it as-is for now and do a larger refactor/review in another PR.